Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```diff
@@ -0,0 +1,355 @@
+        state.pop();
+      }
+      i = i + 1;
+    }
+    finalState = state[state.length - 1];
+    return { matched: flag, state: finalState };
+  }
+
+  protected completion(params: TextDocumentPositionParams): Thenable<CompletionList | null> {
+    const document = this.documents.get(params.textDocument.uri);
+    if (!document) {
+      return Promise.resolve(null);
+    }
+    const text = this.getLGDocumentContent(document);
+    let templates: LGTemplate[] = [];
+    try {
+      const lgResources = getLGResources(text);
+      templates = lgResources.Templates;
+      // eslint-disable-next-line no-empty
+    } catch (error) {}
+    const completionList: CompletionItem[] = [];
+    templates.forEach(template => {
+      const item = {
+        label: template.Name,
+        kind: CompletionItemKind.Reference,
+        insertText:
+          template.Parameters.length > 0 ? template.Name + '(' + template.Parameters.join(', ') + ')' : template.Name,
+        documentation: template.Body,
+      };
+      completionList.push(item);
+    });
+
+    buildInfunctionsMap.forEach((value, key) => {
+      const item = {
+        label: key,
+        kind: CompletionItemKind.Function,
+        insertText: key + '(' + this.removeParamFormat(value.Params.toString()) + ')',
+        documentation: value.Introduction,
+      };
+      completionList.push(item);
+    });
+
+    const matchResult = this.matchedStates(params);
+    // TODO: more precise match
+    if (
+      matchResult &&
+      matchResult.matched &&
+      matchResult.state &&
+      allowedCompletionStates.includes(matchResult.state.toLowerCase())
+    ) {
+      return Promise.resolve({ isIncomplete: true, items: completionList });
+    } else {
+      return Promise.resolve(null);
+    }
+  }
+
+  protected validate(document: TextDocument): void {
+    this.cleanPendingValidation(document);
+    setTimeout(() => {
+      this.pendingValidationRequests.delete(document.uri);
+      this.doValidate(document);
+    });
+  }
+
+  protected cleanPendingValidation(document: TextDocument): void {
+    const request = this.pendingValidationRequests.get(document.uri);
+    if (request !== undefined) {
+      clearTimeout(request);
+      this.pendingValidationRequests.delete(document.uri);
+    }
+  }
+
+  protected doValidate(document: TextDocument): void {
+    let text = document.getText();
+    let lineOffset = 0;
+    const LGDocument = this.getLGDocument(document);
+    if (LGDocument) {
+      // concat new content for validate
+      const { content, template } = LGDocument;
+      const updatedTemplate = {
+        Name: template.Name,
+        Parameters: template.Parameters,
+        Body: text,
+      };
+      try {
+        text = updateTemplateInContent(content, updatedTemplate);
+        lineOffset = getTemplatePositionOffset(content, updatedTemplate);
+        // eslint-disable-next-line no-empty
+      } catch (error) {}
```

Comment:
Please put a comment on why silencing errors. It give next developers some insights on what actually happened.